_posts/2024-10-28-using-maven-wildfly-plugin-with-postgresql-ds.adoc

```diff
@@ -0,0 +1,295 @@
+----
+
+So the feature pack and the layer will prepare the provisioned WildFly server to connect to the PostgreSQL driver, and then we can configure the datasource in the server. In the above example, there are scripts that will add the PostgreSQL datasourcefootnote:[https://github.com/jberet/jberet-examples/blob/main/deployment/add-postgresql-ds.cli]:
+
+[source,txt]
+----
+xa-data-source add --name=batch_db --enabled=true --use-java-context=true --use-ccm=true --jndi-name=java:jboss/jsr352/batch_db --xa-datasource-properties={"URL"=>"jdbc:postgresql://localhost:5432/batch_db"} --driver-name=postgresql --password=123 --user-name=batch_user --same-rm-override=false --no-recovery=true
+/subsystem=batch-jberet/jdbc-job-repository=batch_db:add(data-source=batch_db)
+/subsystem=batch-jberet/:write-attribute(name=default-job-repository,value=batch_db)
+----
+
+The above configuration add the PostgreSQL datasource I need, and then it configures the batch subsystem to use the datasource as its job store, and is used in the `wildfly-maven-plugin` configuration section:
+
+[source,xml]
+----
+<scripts>
+    <script>${configure.ds.script}</script>
+</scripts>
+----
+
+The `configure.ds.script` property is configured in the `postgresql` profile:
+
+[source,xml]
+----
+<profile>
+    <id>postgres</id>
+    <activation>
+        <property>
+            <name>postgres</name>
+        </property>
+    </activation>
+    <properties>
+        <configure.ds.script>${project.basedir}/add-postgresql-ds.cli</configure.ds.script>
+    </properties>
+</profile>
+----
+
+With the script execution during the server startup, I can see from the server log that the datasource is bound:
+
+[source,txt]
+----
+17:47:36,063 INFO  [org.jboss.as.connector.subsystems.datasources] (MSC service thread 1-6) WFLYJCA0001: Bound data source [java:jboss/jsr352/batch_db]
+----
+
+Please note that the `script` section configuration can be used with the `provision` goal(and it can be used in other goals as well like `dev`):
+
+[source,xml]
+----
+<execution>
+    <goals>
+        <goal>provision</goal>
+    </goals>
+</execution>
+----
+
+If you want to use the `script` with the `package` goal:
+
+[source,xml]
+----
+<execution>
+    <goals>
+        <goal>package</goal>
+    </goals>
+</execution>
+----
+
+Then you should configure the script like this:
+
+[source,xml]
+----
+<packaging-scripts>
+    <packaging-script>
+        <scripts>
+            <script>${configure.ds.script}</script>
+        </scripts>
+        <!-- Expressions resolved during server execution -->
+        <resolve-expressions>false</resolve-expressions>
+    </packaging-script>
+</packaging-scripts>
+----
+
+Here is an example that is showing the above configuration method:
+
+* https://github.com/wildfly/quickstart/blob/main/jaxrs-jwt/pom.xml#L177-L185[quickstart/jaxrs-jwt/pom.xml at main · wildfly/quickstart]
```

Comment:
We should copy/paste an example or point to an example like https://docs.wildfly.org/wildfly-maven-plugin/releases/5.0/package-mojo.html#packagingScripts. This link could become invalid if the quickstart goes away or the POM is refactoried.